plugins/openselected.lua

```diff
@@ -31,16 +31,34 @@ config.plugins.openselected = common.merge({
   }
 }, config.plugins.openselected)
 
+local function select_word_under_cursor(doc)
+  local line, col = doc:get_selection() -- will be cursor if no selection
+  local text = doc.lines[line]
+  local pattern = "%s'\""
+
+  if not text or text == "\n" then return nil end
+
+  local start_col = text:sub(1,col):find("[^"..pattern.."]")
+  local end_col = col + text:sub(col, #text):find("["..pattern.."]") - 1
```

Comment:
Instead of splitting the string, you can use the second parameter of `find` to specify where to start.